### PR TITLE
to see

### PR DESCRIPTION
**Pull request by taylorz1**

### CODE REVIEW

File: src/handler/mongoHandler.js [hunk 1]
```diff
@@ -61,20 +61,13 @@ class MongoHandler {
...
       .then((response) => {
-        const body = {
-            uuid: content.uuid,
-            action: content.action
-        };
         if (response.upsertedCount === 0) {
           this.logger.info(`[ ${logTag} ] wish ${content.name} by ${content.wisher} already exists`);
-          body.statusCode = 409;
-          body.message = "Wish by that name and author already exists. Please try another.";
+          resolve(false);
```
Comment (lamdav) [inline src/handler/mongoHandler.js]: This will requeue the create.

File: src/handler/mongoHandler.js [hunk 2]
```diff
@@ -95,18 +88,12 @@ class MongoHandler {
...
       .then((response) => {
-        const body = {
-            uuid: content.uuid,
-            action: content.action
-        };
         if (response.modifiedCount === 0) {
           this.logger.info(`[ ${logTag} ] wish ${content.name} by ${content.wisher} doesn't exist with id ${content.wishId}`);
-          body.statusCode = 409;
-          body.message = "Wish by that ID does not exist. Please try another.";
+          resolve(false);
```
Comment (lamdav) [inline src/handler/mongoHandler.js:138]: Same here.

File: src/handler/mongoHandler.js [hunk 3]
```diff
@@ -95,18 +88,12 @@ class MongoHandler {
     const options = {upsert: false};
     coursedb.updateOne(searchQuery, updateQuery, options)
       .then((response) => {
-        const body = {
-            uuid: content.uuid,
-            action: content.action
-        };
         if (response.modifiedCount === 0) {
           this.logger.info(`[ ${logTag} ] wish ${content.name} by ${content.wisher} doesn't exist with id ${content.wishId}`);
```
Comment (lamdav) [inline src/handler/mongoHandler.js:137]: doesn't -> does not

File: src/handler/mongoHandler.js [hunk 4]
```diff
@@ -122,18 +109,12 @@ class MongoHandler {
...
       .then((response) => {
-        const body = {
-            uuid: content.uuid,
-            action: content.action
-        };
         if (response.deletedCount === 0) {
           this.logger.info(`[ ${logTag} ] Wish with id ${content.wishId} does not exist`);
-          body.statusCode = 409;
-          body.message = "Wish by that ID does not exist, please try again.";
+          resolve(false);
```
Comment (lamdav) [inline src/handler/mongoHandler.js:159]: Same here.

**Review by lamdav — CHANGES_REQUESTED** [review]
I would rather generalize the `handle` methods. So that we don't just copy and paste the same base code and change 2-3 lines for every query.

File: src/handler/mongoHandler.js (hunk 1 above)
Comment (taylorz1) [reply src/handler/mongoHandler.js]: Should it just be resolve(true) then? Do we just assume the data is lost?

Comment (lamdav) [reply src/handler/mongoHandler.js]: If it is false, it would just be requeued. Maybe we should make wish create also synchronous.

**Comment by lamdav** [comment]
I went ahead a made `wishCreate` synchronous. You can handle it how it was handled before using `uuid` and `respond`. The `update` can remain how it is.

File: src/handler/mongoHandler.js [hunk 5]
```diff
@@ -47,6 +47,44 @@ class MongoHandler {
...
 
+  handleMongoCreate(db, content, resolve, logTag, searchQuery, updateQuery, callback) {
+    const logTag = logTag
+    this.logger.info(`[ ${logTag} ] handling ${content.action}`);
+    const options = {upsert: true};
+    coursedb.updateOne(searchQuery, updateQuery, options)
+      .then((response) => {
+        if (response.upsertedCount === 0) {
+          this.logger.info(`[ ${logTag} ] failMessage`);
+          resolve(false);
```
Comment (lamdav) [inline src/handler/mongoHandler.js:58]: All create flow should be using the calling `respond` right.

File: src/handler/mongoHandler.js [hunk 6]
```diff
@@ -47,6 +47,44 @@ class MongoHandler {
...
   }
 
+  handleMongoCreate(db, content, resolve, logTag, searchQuery, updateQuery, callback) {
+    const logTag = logTag
+    this.logger.info(`[ ${logTag} ] handling ${content.action}`);
+    const options = {upsert: true};
+    coursedb.updateOne(searchQuery, updateQuery, options)
+      .then((response) => {
+        if (response.upsertedCount === 0) {
+          this.logger.info(`[ ${logTag} ] failMessage`);
```
Comment (lamdav) [inline src/handler/mongoHandler.js:57]: I thought we agreed it wasn't worth generalizing and abstracting create workflow.

Comment (lamdav) [reply src/handler/mongoHandler.js:57]: If you are, the reason why we generalize is to not have obscure messages such as `failMessage`

File: src/handler/mongoHandler.js [hunk 7]
```diff
@@ -47,6 +47,44 @@ class MongoHandler {
     });
   }
 
+  handleMongoCreate(db, content, resolve, logTag, searchQuery, updateQuery, callback) {
+    const logTag = logTag
+    this.logger.info(`[ ${logTag} ] handling ${content.action}`);
+    const options = {upsert: true};
+    coursedb.updateOne(searchQuery, updateQuery, options)
```
Comment (lamdav) [inline src/handler/mongoHandler.js:54]: Where are you getting `coursedb`?

Comment (lamdav) [reply src/handler/mongoHandler.js:54]: Javascript isn't dynamically scoped from what I know unless `coursedb` is global which is a bad idea.

File: src/handler/mongoHandler.js [hunk 8]
```diff
@@ -47,6 +47,44 @@ class MongoHandler {
...
+      .then((response) => {
+        if (response.upsertedCount === 0) {
+          this.logger.info(`[ ${logTag} ] failMessage`);
+          resolve(false);
+        } else {
+          this.logger.info(`[ ${logTag} ] passMessage`);
```
Comment (lamdav) [inline src/handler/mongoHandler.js:60]: What is `passMessage`? If you got an error and needed to debug this, do you want to see generic messages such as `passMessage` and `failMessage`?

File: src/handler/mongoHandler.js [hunk 9]
```diff
@@ -47,6 +47,44 @@ class MongoHandler {
...
+      .catch((error) => {
+        this.logger.error(`[ ${logTag} ] Mongo failed update query: ${JSON.stringify(error.message)}`);
+        resolve(false);
+      });
+  }
+
+  handleMongoRemove(db, content, resolve, logTag, searchQuery, failMessage, passMessage) {
+      const logTag = logTag
+      this.logger.info(`[ ${logTag} ] handling ${content.action}`);
+      coursedb.deleteOne(searchQuery)
```
Comment (lamdav) [inline src/handler/mongoHandler.js:73]: What is `coursedb`?

File: src/handler/mongoHandler.js [hunk 10]
```diff
@@ -47,6 +47,44 @@ class MongoHandler {
...
+          resolve(true);
+        }
+      })
+      .catch((error) => {
+        this.logger.error(`[ ${logTag} ] Mongo failed update query: ${JSON.stringify(error.message)}`);
+        resolve(false);
+      });
+  }
+
+  handleMongoRemove(db, content, resolve, logTag, searchQuery, failMessage, passMessage) {
```
Comment (lamdav) [inline src/handler/mongoHandler.js:70]: Are we going to generalize remove as well?

File: src/handler/mongoHandler.js [hunk 11]
```diff
@@ -47,6 +47,44 @@ class MongoHandler {
...
+      const logTag = logTag
+      this.logger.info(`[ ${logTag} ] handling ${content.action}`);
+      coursedb.deleteOne(searchQuery)
+        .then((response) => {
+          if (response.deletedCount === 0) {
+            this.logger.info(`[ ${logTag} ] failMessage`);
```
Comment (lamdav) [inline src/handler/mongoHandler.js:76]: This seems like a case of poor copying and pasting. The variable `failMessage` is defined here but not in the `handeMongoCreate`. 

If you want to pass in a fail message, you need to do `${failMessage}`. This is just special javascript syntax to essentially do `"[ " + logTag + " ] " + failMessage`

File: src/handler/mongoHandler.js [hunk 12]
```diff
@@ -47,6 +47,44 @@ class MongoHandler {
...
+          } else {
+            this.logger.info(`[ ${logTag} ] passMessage`);
```
Comment (lamdav) [inline src/handler/mongoHandler.js:78]: Same here.

**Review by lamdav — CHANGES_REQUESTED** [review]

File: src/handler/mongoHandler.js [hunk 13]
```diff
@@ -47,6 +47,44 @@ class MongoHandler {
     });
   }
 
+  handleMongoCreate(db, content, resolve, logTag, searchQuery, updateQuery, callback) {
+    const logTag = logTag
```
Comment (zhoutwo) [inline src/handler/mongoHandler.js:51]: Please add a semicolon after all statements

**Review by zhoutwo — CHANGES_REQUESTED** [review]

**Comment by zhoutwo** [comment]
Also, the title of the PR should be reflective of what the changes do.

**Comment by zhoutwo** [comment]
And it seems certain files have conflict. Please rebase with master.

**Pull request closed without merging**